src/client/gfs_client.h

```diff
@@ -0,0 +1,44 @@
+#ifndef CLIENT_GFS_CLIENT_H
+#define CLIENT_GFS_CLIENT_H
+
+#include "google/protobuf/stubs/status.h"
+#include "google/protobuf/stubs/statusor.h"
+
+namespace gfs {
+namespace client {
+
+// Use the Status and Statusor objects provided by protobuf
+using google::protobuf::util::Status;
+using google::protobuf::util::StatusOr;
+
+/* Data type for the return of read operation. This includes how many bytes have been
+ * read and a pointer of the buffer */
+struct Data {
+   int bytesRead;
+   void* buffer; 
+};
+
+/* Provide an interface function to let user to initialize client and configure
+ * parameters such as master's url. A user should call this function before
+ * performing any of the following functions as initializations are needed */
+void configClient(const char* masterUrl);
+
+Status create(const char* pathname);
+Status remove(const char* pathname); 
+
+/* Let's plan to support the following open mode:
+ * Read Mode | Write Mode | Create Mode | Is Directory
+ * Possible combination: 
+ * Create | Write - create and write to a file
+ * Create | Is Directory - Create a direcotry 
+ */
+Status open(const char* pathname, int flags);
+Status close(const char* pathname);
+
+StatusOr<Data> read(const char* pathname, size_t offset, size_t nbytes);
+Status write(const char* path, void* buffer, size_t offset, size_t nbytes);
+
+} /* namespace client */
```

Comment:
nit. michael and I have been using `// namespace xxx`. can we standarlize this

Reply:
I dont think you addressed this issue yet @xicheng87

Reply:
I did - the name space style has been modified in the last change.